pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go

```diff
@@ -235,32 +250,87 @@ func nodesWherePreemptionMightHelp(nodes []*framework.NodeInfo, m framework.Node
 	return potentialNodes
 }
 
+// calculateNumCandidates returns the number of candidates the findCandidates
+// method must produce from dry running based on the constraints given by
+// <maxCandidateNodesPercentage> and <minCandidates>. The number of candidates
+// returned will never be greater than <numNodes>.
+func (pl *DefaultPreemption) calculateNumCandidates(numNodes int32) int32 {
+	n := (numNodes * pl.args.MaxCandidateNodesPercentage) / 100
+	if n < pl.args.MinCandidates {
+		n = pl.args.MinCandidates
+	}
+	if n > numNodes {
+		n = numNodes
+	}
+	return n
+}
+
+// calculateOffset returns an <offset> and <numCandidates> that the caller must
+// use to find candidates. A variable offset is used as the starting index when
+// dry running preemption so that the same set of nodes don't get preempted
+// every time. If <offset> + <numCandidates> runs past the number of nodes, the
+// caller is expected to wrap around the node list.
+func (pl *DefaultPreemption) calculateOffset(numNodes int32) (int32, int32) {
```

Comment:
According to the function, it just returns `numCandidates`  and previous `pl.candidateOffset`. Would it be better to  get offset from `pl.candidateOffset` and rename the function name to `updateOffset()`?

Reply:
Done (renamed to updateAndReturnOffset), but it's simpler to have this function return the correct offset than use candidateOffset directly because of some corner cases around the zero value.